ludwig/automl/base_config.py

```diff
@@ -222,20 +222,23 @@ def get_dataset_info(df: Union[pd.DataFrame, dd.core.DataFrame]) -> DatasetInfo:
 
 
 def is_field_boolean(source: DataSource, field: str) -> bool:
-    num_unique_values, unique_values, _ = source.get_distinct_values(field, max_values_to_return=4)
+    """Returns a boolean indicating whether the object field should have a bool dtype.
+
+    Columns with object dtype that have 3 distinct values of which one is Nan/None is a bool type column.
+    """
+    unique_values = source.df[field].unique()
+    num_unique_values = len(unique_values)
     if num_unique_values <= 3:
-        for entry in unique_values:
+        if None in unique_values:
+            return True
+        for unique_value in unique_values:
             try:
-                if np.isnan(entry):
-                    continue
+                if np.isnan(unique_value):
+                    return True
```

Comment:
Also why are we returning True here?

Reply:
Because one of the three distinct values is a NaN, which means that the others are valid, which therefore makes this a valid bool type